### PR TITLE
[Feature] Support configure rocksdb meta column family options.

### LINKED ISSUE #14395: Should support configure rocksdb meta column family options.
## Feature request

**Is your feature request related to a problem? Please describe.**

the default rocksdb block cache size is 8M, maybe too small, it should can be changed.
the other column family options should be a configurable string.

**Describe the solution you'd like**

add a configuration to delivery it to rocksdb.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Linkerist**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #14395

## Problem Summary(Required) ：

The default rocksdb block cache size is 8M, which may be too small. It should be configured.
just like block cache size, other column family options should also be configured.
This PR adds a configuration in be.conf to deliver a string to rocksdb `GetColumnFamilyOptionsFromString` API to parse.
Besides, prefix_extractor and compression are fixed options in case of incorrect.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3
  - [x] 2.2

### CODE REVIEW

File: be/src/common/config.h [hunk 1]
```diff
@@ -895,4 +895,6 @@ CONF_Int32(exception_stack_level, "1");
 CONF_String(exception_stack_white_list, "std::");
 CONF_String(exception_stack_black_list, "apache::thrift::,ue2::,arangodb::");
 
+CONF_Int32(rocksdb_block_cache, "512"); // 512MB
```
Comment (howrocks) [inline be/src/common/config.h:898]: @Linkerist you should add a general config item to support all rocksdb options，not just for block cache size.

**Review by mark-phoenix-ai — CHANGES_REQUESTED** [review]

File: be/src/common/config.h (hunk 1 above)
Comment (Linkerist) [reply be/src/common/config.h]: changed.

File: be/src/common/config.h [hunk 2]
```diff
@@ -895,4 +895,8 @@ CONF_Int32(exception_stack_level, "1");
 CONF_String(exception_stack_white_list, "std::");
 CONF_String(exception_stack_black_list, "apache::thrift::,ue2::,arangodb::");
 
+CONF_String(rocksdb_meta_cf_opts_string,
```
Comment (chaoyli) [inline be/src/common/config.h:898]: Add a clear message about the problem, and solution, and result in the commit message

Comment (howrocks) [inline be/src/common/config.h:898]: The config name is too long and not clear.

File: be/src/common/config.h [hunk 3]
```diff
@@ -895,4 +895,8 @@ CONF_Int32(exception_stack_level, "1");
...
+            "prefix_extractor=rocksdb.FixedPrefix.4;compression=kSnappyCompression;block_based_table_factory={block_"
+            "cache=128M}");
```
Comment (decster) [inline be/src/common/config.h:900]: just "block_based_table_factory={block_cache=128M}"

File: be/src/storage/kv_store.cpp [hunk 4]
```diff
@@ -79,8 +83,7 @@ Status KVStore::init(bool read_only) {
     cf_descs[1].name = STARROCKS_COLUMN_FAMILY;
     cf_descs[1].options.compression = rocksdb::kSnappyCompression;
     cf_descs[2].name = META_COLUMN_FAMILY;
-    cf_descs[2].options.prefix_extractor.reset(NewFixedPrefixTransform(PREFIX_LENGTH));
-    cf_descs[2].options.compression = rocksdb::kSnappyCompression;
```
Comment (decster) [inline be/src/storage/kv_store.cpp:83]: this should keep been set and not changed by user config.

**Comment by Linkerist** [comment]
run starrocks_be_unittest

**Review by mark-phoenix-ai — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Linkerist** [comment]
run starrocks_be_unittest

**Review by decster — APPROVED** [review]

**Comment by Linkerist** [comment]
run starrocks_admit_test

File: be/src/common/config.h [hunk 5]
```diff
@@ -895,4 +895,6 @@ CONF_Int32(exception_stack_level, "1");
 CONF_String(exception_stack_white_list, "std::");
 CONF_String(exception_stack_black_list, "apache::thrift::,ue2::,arangodb::");
 
+CONF_String(rocksdb_cf_options_string, "block_based_table_factory={block_cache=128M}");
```
Comment (chaoyli) [inline be/src/common/config.h:898]: Add a concise message about the feature and solution.

**Comment by Linkerist** [comment]
run starrocks_admit_test

**Comment by Linkerist** [comment]
run starrocks_admit_test

**Comment by Linkerist** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Comment by Linkerist** [comment]
@Mergifyio backport branch-2.5

**Comment by Linkerist** [comment]
@Mergifyio backport branch-2.4

**Comment by Linkerist** [comment]
@Mergifyio backport branch-2.3

**Comment by Linkerist** [comment]
@Mergifyio backport branch-2.2

**Pull request merged**